File: csharp/1486.xor-operation-in-an-array.cs
/*
 * @lc app=leetcode id=1486 lang=csharp
 *
 * [1486] XOR Operation in an Array
 */

using System.Linq;

// @lc code=start
public class Solution {
    public int XorOperation(int n, int start) {
        return Enumerable
            .Range(0, n)
            .Select(i => start + 2 * i)
            .Aggregate((acc, x) => acc ^ x);
    }
}
// @lc code=end

File: java/1486.xor-operation-in-an-array.java
/*
 * @lc app=leetcode id=1486 lang=java
 *
 * [1486] XOR Operation in an Array
 */

// @lc code=start
class Solution {
    public int xorOperation(int n, int start) {
        int ret = 0;
        for (int i = 0; i < n; ++i) {
            ret ^= (start + 2 * i);
        }
        return ret;
    }
}
// @lc code=end

